### ISSUE TITLE
cargo vet diff shows diffs of files that aren't packaged

### ISSUE THREAD

**Issue by glandium** [mozilla/cargo-vet#182]
For instance, `cargo vet diff semver 1.0.9 1.0.10` shows differences for `.cargo_vcs_info.json`, `.github/workflows/ci.yml`, `Cargo.toml.orig`. None of these files are (presumably) shipped by `cargo publish`, and they're not pulled by `cargo vendor`.

**Comment by Gankra** [comment]
These files are in fact shipped by `publish`: https://doc.rust-lang.org/cargo/commands/cargo-package.html

I agree it would be nice to filter them out, I couldn't see an obvious way to do this with our current diff backend just being "call git diff --no-index", and I would prefer to not mutate the directories during diffing.

**Comment by glandium** [comment]
`.cargo_vcs_info.json` might, but probably not the others.

**Comment by Gankra** [comment]
We're literally downloading the `.crate` files from crates.io, and those are just `.tar.gz` files that we unpack. The SHAs in the registry are for those `.crate` files, so that set of files is the canonical publish.

If you see it in the diff, that's something published to crates.io. 

`Cargo.toml.orig` exists because cargo rewrites your Cargo.toml on publish, per

> [patch], [replace], and [workspace] sections are removed from the manifest.

**Comment by Gankra** [comment]
Here's some of the logic cargo does to "clean up" the dir when you vendor:

https://github.com/rust-lang/cargo/blob/6b8e1922261bbed1894bf40069fb2d5dc8d62fb0/src/cargo/ops/vendor.rs#L322-L341

**Comment by Gankra** [comment]
(the relevant fixme/code)

https://github.com/mozilla/cargo-vet/blob/24d8d8d1e9cd4264b64ee3e79ef54301a0679c21/src/main.rs#L1418-L1426

**Comment by bholley** [comment]
Yeah the `cargo publish` behavior here surprising but I think our job is to show the package contents as they are.
